### PR TITLE
Replacement of getName with lookupPath

### PR DESCRIPTION
**Pull request by peldax**
Changes:
- new protected variable in DataGrid class `componentFullName`, which contains result of `$this->lookupPath()`.
- replaced occurrences of `$this->getName()` with `$this->componentFullName` in DataGrid class and in datagrid.latte

By doing this, datagrid works with its full and unique name.

It is solution to #538 , where link was generated incorrectly, because the component's name was not full.

It might fix some other issues, because some JS selectors are currently using componenets name.

### CODE REVIEW

**Comment by peldax** [comment]
I also noted, someone recommended this update few year ago on Nette forum and ended up unnoticed.

https://forum.nette.org/cs/25444-ublaboo-datagrid-mocny-rychly-rozsiritelny-hezky-anglicky-dokumentovany-datagrid?p=3#p171391

**Comment by paveljanda** [comment]
👍 I like it

File: src/DataGrid.php [hunk 1]
```diff
@@ -2363,6 +2369,8 @@ public function sendNonEmptyFiltersInPayload($filters)
 			}
 		}
 
+        bdump($non_empty_filters);
```
Comment (f3l1x) [inline src/DataGrid.php]: Could you please remove this?

**Comment by peldax** [comment]
@f3l1x dump odstatněn, nechápu jakto že mi tam zůstal, díky za review

**Comment by f3l1x** [comment]
Muzes to prosim @peldax rebasnout na aktualni v5.x? Pak to radi mergneme.

**Comment by peldax** [comment]
@f3l1x Mělo by být, prosím mrkněte na to co nejdříve :+1:

**Comment by paveljanda** [comment]
@f3l1x But it is a BC break, it has to be released in new major version.

**Comment by MartinSzollos2016** [comment]
@peldax Could you please fix GroupActionCollection.php(121 -> 126)
`		$container['group_action']->addCondition(Form::FILLED)
			->toggle(strtolower($this->datagrid->getFullName()) . 'group_action_submit');

		$container->addSubmit('submit', 'ublaboo_datagrid.execute')
			->setValidationScope([$container])
			->setAttribute('id', strtolower($this->datagrid->getFullName()) . 'group_action_submit');`

**Comment by peldax** [comment]
@MartinSzollos2016 done, good find.

**Comment by peldax** [comment]
@paveljanda @f3l1x Would you merge this PR in 6.0 ?

File: composer.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-	"name": "ublaboo/datagrid",
+	"name": "peldax/datagrid",
```
Comment (paveljanda) [inline composer.json]: Could you please revert this? :)

Comment (peldax) [reply composer.json]: It is removed in the patch-2 branch as linked in my comment.

**Comment by peldax** [comment]
I've already rebased once in October... I am gonna make whole new PR with correct branch and rebase.

**Comment by peldax** [comment]
Created new PR from fresh branch and rebased on master:

https://github.com/contributte/datagrid/pull/787

**Pull request closed without merging**